format_test.go

```diff
@@ -0,0 +1,29 @@
+package dns
+
+import (
+	"testing"
+)
+
+func TestFieldEmptyAData(t *testing.T) {
+	rr := &A{
+		Hdr:  RR_Header{},
```

Comment:
one nit: These tests could just be `Field(new(A), 1)` and `Field(new(AAAA), 1)`. It would probably be worth merging them into a single test case.

Reply:
+1, can you make that change? Then I'll merge.

Reply:
I don't fully agree. I specifically didn't want to test the nil header for the because I think that should go in a separate test if that should be tested.

I made two separate tests because they test two different flows in the Field function.